apps/studio/src/pages/sites/[siteId]/settings/logo.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+import { Box } from "@chakra-ui/react"
+import { useToast } from "@opengovsg/design-system-react"
+import { LogoSettingsSchema } from "@opengovsg/isomer-components"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+import { isEqual } from "lodash"
+import { BiPaint } from "react-icons/bi"
+
+import type { NextPageWithLayout } from "~/lib/types"
+import { PermissionsBoundary } from "~/components/AuthWrappers"
+import {
+  SettingsEditorGridItem,
+  SettingsGrid,
+  SettingsPreviewGridItem,
+} from "~/components/Settings"
+import {
+  BRIEF_TOAST_SETTINGS,
+  SETTINGS_TOAST_MESSAGES,
+} from "~/constants/toast"
+import { EditSettingsPreview } from "~/features/editing-experience/components/EditSettingsPreview"
+import { ErrorProvider } from "~/features/editing-experience/components/form-builder/ErrorProvider"
+import FormBuilder from "~/features/editing-experience/components/form-builder/FormBuilder"
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
+import { siteSchema } from "~/features/editing-experience/schema"
+import { SettingsEditingLayout } from "~/features/settings/SettingsEditingLayout"
+import { SettingsHeader } from "~/features/settings/SettingsHeader"
+import { useNavigationEffect } from "~/hooks/useNavigationEffect"
+import { useNewSettingsPage } from "~/hooks/useNewSettingsPage"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { LogoSettings, logoSettingsValidator } from "~/schemas/site"
+import { SiteSettingsLayout } from "~/templates/layouts/SiteSettingsLayout"
+import { trpc } from "~/utils/trpc"
+
+const LogoSettingsPage: NextPageWithLayout = () => {
+  const { siteId: rawSiteId } = useQueryParse(siteSchema)
+  const siteId = Number(rawSiteId)
+  const router = useRouter()
+  const isEnabled = useNewSettingsPage()
+  const trpcUtils = trpc.useUtils()
+  const toast = useToast(BRIEF_TOAST_SETTINGS)
+  const [{ logoUrl, favicon, ...rest }] = trpc.site.getConfig.useSuspenseQuery({
+    id: siteId,
+  })
+  const [{ name: siteName }] = trpc.site.getSiteName.useSuspenseQuery({
+    siteId,
+  })
```

Comment:
nit: can get it from `getConfig` query above to save one query. Also then no need for query invalidation onsuccess for this